com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/HDRenderPipeline.VolumetricClouds.cs

```diff
@@ -0,0 +1,684 @@
+            InitializeVolumetricCloudsMap();
+            InitializeVolumetricCloudsShadows();
+        }
+
+        void ReleaseVolumetricClouds()
+        {
+            if (!m_Asset.currentPlatformRenderPipelineSettings.supportVolumetricClouds)
+                return;
+
+            // Release the additional sub components
+            ReleaseVolumetricCloudsMap();
+            ReleaseVolumetricCloudsShadows();
+        }
+
+        void AllocatePresetTextures()
+        {
+            // Build our default cloud map
+            m_SparsePresetMap = new Texture2D(1, 1, GraphicsFormat.R8G8B8A8_UNorm, TextureCreationFlags.None) { name = "Default Sparse Texture" };
+            m_SparsePresetMap.SetPixel(0, 0, new Color(0.9f, 0.0f, 0.0625f, 1.0f));
+            m_SparsePresetMap.Apply();
+
+            m_CloudyPresetMap = new Texture2D(1, 1, GraphicsFormat.R8G8B8A8_UNorm, TextureCreationFlags.None) { name = "Default Cloudy Texture" };
+            m_CloudyPresetMap.SetPixel(0, 0, new Color(0.9f, 0.2f, 0.0625f, 1.0f));
+            m_CloudyPresetMap.Apply();
+
+            m_OvercastPresetMap = new Texture2D(1, 1, GraphicsFormat.R8G8B8A8_UNorm, TextureCreationFlags.None) { name = "Default Overcast Texture" };
+            m_OvercastPresetMap.SetPixel(0, 0, new Color(0.5f, 0.0f, 0.375f, 1.0f));
+            m_OvercastPresetMap.Apply();
+
+            m_StormCloudsPresetMap = new Texture2D(1, 1, GraphicsFormat.R8G8B8A8_UNorm, TextureCreationFlags.None) { name = "Default Storm Texture" };
+            m_StormCloudsPresetMap.SetPixel(0, 0, new Color(1.0f, 0.8f, 0.375f, 1.0f));
+            m_StormCloudsPresetMap.Apply();
+        }
+
+        // Function that fills the buffer with the ambient probe values
+        unsafe void SetPreconvolvedAmbientLightProbe(ref ShaderVariablesClouds cb, HDCamera hdCamera, VolumetricClouds settings)
+        {
+            SphericalHarmonicsL2 probeSH = SphericalHarmonicMath.UndoCosineRescaling(m_SkyManager.GetAmbientProbe(hdCamera));
+            probeSH = SphericalHarmonicMath.RescaleCoefficients(probeSH, settings.ambientLightProbeDimmer.value);
+            ZonalHarmonicsL2.GetCornetteShanksPhaseFunction(m_PhaseZHClouds, 0.0f);
+            SphericalHarmonicsL2 finalSH = SphericalHarmonicMath.PremultiplyCoefficients(SphericalHarmonicMath.Convolve(probeSH, m_PhaseZHClouds));
+
+            SphericalHarmonicMath.PackCoefficients(m_PackedCoeffsClouds, finalSH);
+            for (int i = 0; i < 7; i++)
+                for (int j = 0; j < 4; ++j)
+                    cb._AmbientProbeCoeffs[i * 4 + j] = m_PackedCoeffsClouds[i][j];
+        }
+
+        // Allocation of the first history buffer
+        static RTHandle VolumetricClouds0HistoryBufferAllocatorFunction(string viewName, int frameIndex, RTHandleSystem rtHandleSystem)
+        {
+            return rtHandleSystem.Alloc(Vector2.one * 0.5f, TextureXR.slices, colorFormat: GraphicsFormat.R16G16B16A16_SFloat, dimension: TextureXR.dimension,
```

Comment:
questoin:  does it need to be R16G16B16A16_SFloat or can it be 111110f?

Reply:
No can't, alpha channel is used